### PR TITLE
ui: fix deploy vm in basic zone

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes error during VM deployment in a basic zone

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Error
Screenshot from 2022-01-13 13-29-38

### How Has This Been Tested?

Steps to reproduce:
- Have a multi-zone setup with one basic zone
- Open deploy vm wizard
- Select basic zone
- Without selecting any security group, try to deploy VM

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by slavkap — COMMENTED** [review]
LGTM and I did some test

File: ui/src/views/compute/DeployVM.vue [hunk 1]
```diff
@@ -1659,7 +1659,7 @@ export default {
             }
           }
         }
-        if (this.securitygroupids.length > 0) {
+        if (this.securitygroupids && this.securitygroupids.length > 0) {
```
Comment (DaanHoogland) [inline ui/src/views/compute/DeployVM.vue:1662]: question: if above the update method adds an empty list, is this extra check still needed? (or vice versa?)

Comment (utchoang) [reply ui/src/views/compute/DeployVM.vue:1662]: I think it would be better to change it to `if (this.securitygroupids.length > 0) {`

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by utchoang — APPROVED** [review]
LGTM!

**Review by slavkap — APPROVED** [review]

**Pull request merged**